sips/dynamic_qbft_timeout.md

```diff
@@ -0,0 +1,90 @@
+| Author         | Title                 | Category | Status   |
+| -------------- | --------------------- | -------- | -------- |
+| Gal Rogozinski |  QBFT timeout | Core     | approved |
+
+
+**Summary**  
+Changes to the constant timeouts we defined in [SIP#6](https://github.com/bloxapp/SIPs/blob/main/sips/constant_qbft_timeout.md) that should improve performance. 
+
+**Rational**  
+Currently for all duties, we have a `quickTimeout` of 2 seconds before sending a `RoundChange` message and a `slowTimeout` of 2 minutes starting from the 8th round.The timer starts when the QBFT instance starts. This approach has 2 issues:
```

Comment:
Could you clarify btw why there is a need for `slowTimeout` ? Is it just purely for qbft or do we have anything in SSV node that would make use of such long timeout (I think most/all duties must finish execution much faster or just will be missed)

Reply:
@iurii-ssv It was introduced in https://github.com/ssvlabs/SIPs/pull/22/files, attester duties may be submitted within 32 slots/6.4m

Reply:
The idea is simply wait for a very long time if after 8 rounds you didn't reach consensus. This is because we are probably in a very high latency environment

Reply:
> @iurii-ssv It was introduced in https://github.com/ssvlabs/SIPs/pull/22/files, attester duties may be submitted within 32 slots/6.4m

right, 

at any rate, I always thought the **spec** should define an interface (not an implementation) everybody wants to conform to and implement it as they see fit for their use-case/application, meaning:
- QBFT shouldn't have to know anything about duty(s), it should be the other way around - duty configures QBFT Instance to work as it needs to
- different duty types should have different `total round count`, `round 1 timeout`, `round 2 timeout`, ... (seems like these values should not be constants, but rather vars - so it doesn't send the wrong message that "these values are uniform for everyone")

It's not really an issue to worry about, just trying to get some consistency (and knowledge) if possible.